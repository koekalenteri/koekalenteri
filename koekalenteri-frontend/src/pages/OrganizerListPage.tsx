import { CloudSync } from '@mui/icons-material';
import { Button, Stack } from '@mui/material';
import { GridColDef } from '@mui/x-data-grid';
import { Organizer } from 'koekalenteri-shared/model';
import { computed, toJS } from 'mobx';
import { observer } from 'mobx-react-lite';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { QuickSearchToolbar, StyledDataGrid } from '../components';
import { FullPageFlex } from '../layout';
import { useStores } from '../stores';
import { AuthPage } from './AuthPage';

interface OrganizerColDef extends GridColDef {
  field: keyof Organizer
}

export const OrganizerListPage = observer(function OrganizerListPage() {
  const [searchText, setSearchText] = useState('');
  const { t } = useTranslation();
  const { rootStore } = useStores();
  const columns: OrganizerColDef[] = [
    {
      field: 'id',
      headerName: t('id')
    },
    {
      field: 'name',
      headerName: t('name'),
      flex: 2
    }
  ];

  const refresh = async () => {
    rootStore.organizerStore.load(true);
  };

  const rows = computed(() => {
    const lvalue = searchText.toLocaleLowerCase();
    return toJS(rootStore.organizerStore.organizers).filter(o => o.search.includes(lvalue));
  }).get();

  const requestSearch = (searchValue: string) => {
    setSearchText(searchValue);
  };

  return (
    <AuthPage title={t('organizations')}>
      <FullPageFlex>
        <Stack direction="row" spacing={2}>
          <Button startIcon={<CloudSync />} onClick={refresh}>{t('updateData', { data: 'organizations' })}</Button>
        </Stack>

        <StyledDataGrid
          loading={rootStore.organizerStore.loading}
          autoPageSize
          columns={columns}
          components={{ Toolbar: QuickSearchToolbar }}
          componentsProps={{
            toolbar: {
              value: searchText,
              onChange: (event: React.ChangeEvent<HTMLInputElement>) =>
                requestSearch(event.target.value),
              clearSearch: () => requestSearch(''),
            },
          }}
          density='compact'
          disableColumnMenu
          disableVirtualization
          rows={rows}
        />
      </FullPageFlex>
    </AuthPage>
  )
});
