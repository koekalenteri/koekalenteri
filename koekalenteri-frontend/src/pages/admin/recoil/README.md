# State available to authenticated users

These flowcharts describes the relationship between atoms, selectors, effects and persistent storage.

## Events

```mermaid
graph LR
  adminEventsAtom-->remoteAdminEventsEffect
  adminEventsAtom-->storageEffect
  newEventAtom-->storageEffect
  adminShowPastEventsAtom-->storageEffect
  adminEventFilterTextAtom-->storageEffect
  adminEventIdAtom-->storageEffect
  eventClassAtom-->storageEffect
  editableEventByIdAtom-->storageEffect

  storageEffect[(localStorage)]
  remoteAdminEventsEffect-->getEvents[/getEvents/]-->aws

  adminEventSelector-->newEventAtom
  adminEventSelector-->adminEventsAtom

  currentAdminEventSelector-->editableEventByIdAtom

  filteredAdminEventsSelector-->adminEventsAtom
  filteredAdminEventsSelector-->adminEventFilterTextAtom
  filteredAdminEventsSelector-->adminShowPastEventsAtom

  aws[(cloud)]
```

## Officials

```mermaid
graph LR
  officialsAtom-->storageEffect
  officialsAtom-->remoteOfficialsEffect
  officialFilterAtom

  storageEffect[(localStorage)]
  remoteOfficialsEffect-->getOfficials[/getOfficials/]-->aws[(cloud)]

  filteredOfficialsSelector-->officialFilterAtom
  filteredOfficialsSelector-->officialsAtom
```

## Organizers

```mermaid
graph LR
  organizersAtom-->storageEffect
  organizersAtom-->remoteOrganizersEffect
  organizersFilterAtom

  storageEffect[(localStorage)]
  remoteOrganizersEffect-->getOrganizers[/getOrganizers/]-->aws[(cloud)]

  filteredOrganizersSelector-->organizersFilterAtom
  filteredOrganizersSelector-->organizersAtom
```

## Registrations

```mermaid
graph LR
  adminRegistrationIdAtom-->storageEffect
  eventRegistrationsAtom-->storageEffect

  storageEffect[(localStorage)]

  currentEventRegistrationsSelector-->adminEventIdAtom
  currentEventRegistrationsSelector-->eventRegistrationsAtom

  currentEventClassRegistrationsSelector-->eventClassAtom
  currentEventClassRegistrationsSelector-->currentEventRegistrationsSelector

  currentAdminRegistrationSelector-->adminRegistrationIdAtom
  currentAdminRegistrationSelector-->currentEventClassRegistrationsSelector
```
