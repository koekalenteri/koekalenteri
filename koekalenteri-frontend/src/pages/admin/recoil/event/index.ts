export * from "./currentEvent"
export * from "./eventClassAtom"
export * from "./eventSelector"
