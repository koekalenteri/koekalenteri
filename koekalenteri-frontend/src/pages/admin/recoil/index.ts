export * from './emailTemplates'
export * from './events'
export * from './officials'
export * from './organizers'
export * from './registrations'
