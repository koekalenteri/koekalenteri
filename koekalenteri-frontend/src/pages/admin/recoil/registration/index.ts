export * from './currentEventClassRegistrationsQuery'
export * from './currentEventRegistrationsQuery'

export * from './registrationIdAtom'
export * from './registrationQuery'

export * from './registrationsQuery'
