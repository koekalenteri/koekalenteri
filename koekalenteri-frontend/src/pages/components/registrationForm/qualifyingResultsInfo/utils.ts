import type { ManualTestResult, QualifyingResult } from 'koekalenteri-shared/model'
import type { EventResultRequirement, EventResultRequirements, EventResultRequirementsByDate } from '../../../../rules'

import { nanoid } from 'nanoid'

import { unique } from '../../../../utils'
import { objectContains } from '../validation'

const asArray = (v: EventResultRequirements | EventResultRequirement) => (Array.isArray(v) ? v : [v])

function findFirstMissing(requirements: EventResultRequirementsByDate | undefined, results: QualifyingResult[]) {
  if (!requirements) {
    return []
  }
  for (const rule of requirements.rules) {
    for (const opt of asArray(rule)) {
      const { count, ...rest } = opt
      if (results.filter((r) => objectContains(r, rest)).length < count) {
        return rest
      }
    }
  }
}

export function availableTypes(requirements?: EventResultRequirementsByDate) {
  if (!requirements) {
    return []
  }
  return unique(requirements.rules.flatMap((rule) => asArray(rule).map((opt) => opt.type)))
}

export function availableResults(requirements?: EventResultRequirementsByDate, type?: string) {
  if (!requirements) {
    return []
  }
  return unique(
    requirements.rules.flatMap((rule) =>
      asArray(rule)
        .filter((opt) => !type || opt.type === type)
        .map((opt) => (opt.cert ? 'CERT' : opt.result))
    )
  )
}

export function createMissingResult(
  requirements: EventResultRequirementsByDate | undefined,
  results: ManualTestResult[],
  regNo: string
): ManualTestResult {
  const rule = findFirstMissing(requirements, results)
  return {
    id: nanoid(10),
    regNo,
    date: new Date(),
    official: false,
    qualifying: true,
    type: '',
    judge: '',
    location: '',
    result: '',
    class: '',
    ...rule,
  }
}

export function resultBorderColor(qualifying: boolean | undefined) {
  if (qualifying === true) {
    return 'success.light'
  }
  if (qualifying === false) {
    return 'error.main'
  }
}

export function getResultId(result: ManualTestResult | QualifyingResult) {
  if ('id' in result) {
    return result.id
  }
  return result.type + result.class + result.date.toISOString()
}
