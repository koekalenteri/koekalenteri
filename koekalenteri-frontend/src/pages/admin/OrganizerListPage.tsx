import { useCallback } from 'react'
import { useTranslation } from 'react-i18next'
import { CloudSync } from '@mui/icons-material'
import { Button, Stack } from '@mui/material'
import { GridColDef } from '@mui/x-data-grid'
import { Organizer } from 'koekalenteri-shared/model'
import { useRecoilState, useRecoilValue } from 'recoil'

import StyledDataGrid from '../components/StyledDataGrid'
import { filteredOrganizersQuery, organizerFilterAtom, useOrganizersActions } from '../recoil'

import FullPageFlex from './components/FullPageFlex'
import { QuickSearchToolbar } from './components/QuickSearchToolbar'

interface OrganizerColDef extends GridColDef {
  field: keyof Organizer
}

export default function OrganizerListPage() {
  const [searchText, setSearchText] = useRecoilState(organizerFilterAtom)
  const organizers = useRecoilValue(filteredOrganizersQuery)
  const actions = useOrganizersActions()

  const { t } = useTranslation()

  const columns: OrganizerColDef[] = [
    {
      field: 'id',
      headerName: t('id'),
    },
    {
      field: 'name',
      headerName: t('name'),
      flex: 2,
    },
  ]

  const onChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) =>
    setSearchText(event.target.value), [setSearchText])

  const clearSearch = useCallback(() => setSearchText(''), [setSearchText])

  return (
    <>
      <FullPageFlex>
        <Stack direction="row" spacing={2}>
          <Button startIcon={<CloudSync />} onClick={actions.refresh}>{t('updateData', { data: 'organizations' })}</Button>
        </Stack>

        <StyledDataGrid
          columns={columns}
          components={{ Toolbar: QuickSearchToolbar }}
          componentsProps={{
            toolbar: {
              value: searchText,
              onChange,
              clearSearch,
            },
          }}
          rows={organizers}
        />
      </FullPageFlex>
    </>
  )
}
