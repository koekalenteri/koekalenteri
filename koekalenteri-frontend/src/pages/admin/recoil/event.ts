import { EventEx } from "koekalenteri-shared/model"
import { atom, DefaultValue, selector, selectorFamily } from "recoil"

import { getEvent, getEvents } from "../../../api/event"
import { unique, uniqueDate } from "../../../utils"
import { logEffect, storageEffect } from "../../recoil/effects"

export interface DecoratedEvent extends EventEx {
  uniqueClasses: string[]
  uniqueClassDates: (eventClass: string) => Date[]
}

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const currentAdminEvent = selector({
  key: 'currentAdminEvent',
  get: ({ get }) => {
    const eventId = get(adminEventIdAtom)
    return eventId ? get(adminEventSelector(eventId)) : undefined
  },
  set: ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventSelector(newValue.id), newValue)
  },
})

function decorateEvent(event: EventEx): DecoratedEvent {
  return {
    ...event,
    uniqueClasses: unique(event.classes.map(c => c.class)),
    uniqueClassDates: (eventClass: string) => uniqueDate(
      event.classes
        .filter(c => c.class === eventClass)
        .map(c => c.date || event.startDate || new Date()),
    ),
  }
}

export const adminEventsAtom = atom<DecoratedEvent[]>({
  key: 'adminEvents',
  default: getEvents().then(events => events.map(decorateEvent)),
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminEventByIdAtom = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvents/eventId',
  get: (eventId) => async ({ get }) => {
    let event = get(adminEventsAtom).find(event => event.id === eventId)
    if (!event) {
      const fetched = await getEvent(eventId)
      if (fetched) {
        event = decorateEvent(fetched)
      }
    }
    return event
  },
  set: (eventId) => ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventsAtom, oldEvents => {
      const index = oldEvents.findIndex(item => item.id === eventId)
      const newEvents = oldEvents.map(event => ({...event}))
      newEvents.splice(index, 1, newValue)
      return newEvents
    })
  },
})

export const adminEventSelector = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvent',
  get: (eventId) => ({get}) => get(adminEventByIdAtom(eventId)),
  set: (eventId) => ({set}, newValue) => set(adminEventByIdAtom(eventId), newValue),
})

export const eventClassAtom = atom<string | undefined>({
  key: 'eventClass',
  default: selector({
    key: 'eventClass/default',
    get: ({ get }) => get(currentAdminEvent)?.uniqueClasses?.[0],
  }),
  effects: [
    logEffect,
    storageEffect,
  ],
})
