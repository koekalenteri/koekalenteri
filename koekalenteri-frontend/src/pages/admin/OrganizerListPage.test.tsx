import { Suspense } from 'react'
import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { render } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { RecoilRoot } from 'recoil'

import theme from '../../assets/Theme'
import { flushPromisesAndTimers } from '../../test-utils/utils'

import { OrganizerListPage } from './OrganizerListPage'

jest.useFakeTimers()

jest.mock('../../api/organizer')

describe('OrganizerListPage', () => {
  it('renders', async () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <MemoryRouter>
            <Suspense fallback={<div>loading...</div>}>
              <SnackbarProvider>
                <OrganizerListPage />
              </SnackbarProvider>
            </Suspense>
          </MemoryRouter>
        </RecoilRoot>
      </ThemeProvider>,
    )
    await flushPromisesAndTimers()
    expect(container).toMatchSnapshot()
  })
})
