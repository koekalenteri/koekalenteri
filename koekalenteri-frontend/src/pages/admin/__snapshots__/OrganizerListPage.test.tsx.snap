// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrganizerListPage renders 1`] = `
<div>
  <div
    class="MuiBox-root css-1uovp08"
  >
    <div
      class="MuiBox-root css-1tyz8pc"
    >
      <div
        class="MuiStack-root css-1h12o0g-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-16m0f1y-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="CloudSyncIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M21.5 14.98c-.02 0-.03 0-.05.01C21.2 13.3 19.76 12 18 12c-1.4 0-2.6.83-3.16 2.02C13.26 14.1 12 15.4 12 17c0 1.66 1.34 3 3 3l6.5-.02c1.38 0 2.5-1.12 2.5-2.5s-1.12-2.5-2.5-2.5zM10 4.26v2.09C7.67 7.18 6 9.39 6 12c0 1.77.78 3.34 2 4.44V14h2v6H4v-2h2.73C5.06 16.54 4 14.4 4 12c0-3.73 2.55-6.85 6-7.74zM20 6h-2.73c1.43 1.26 2.41 3.01 2.66 5h-2.02c-.23-1.36-.93-2.55-1.91-3.44V10h-2V4h6v2z"
              />
            </svg>
          </span>
          updateData
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        aria-colcount="2"
        aria-multiselectable="false"
        aria-rowcount="3"
        class="MuiDataGrid-root MuiDataGrid-root--densityCompact css-6b2lsp-MuiDataGrid-root"
        role="grid"
      >
        <div>
          <div>
            <div
              class="MuiStack-root css-114556u-MuiStack-root"
            >
              <div
                class="MuiDataGrid-toolbarContainer css-qdjmts-MuiDataGrid-toolbarContainer"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1pupl87-MuiFormControl-root-MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart MuiInputBase-adornedEnd css-dlnixt-MuiInputBase-root-MuiInput-root"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                      data-testid="SearchIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                      />
                    </svg>
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiInput-input MuiInputBase-inputAdornedStart MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
                      id=":r0:"
                      placeholder="searchPlaceholder"
                      type="text"
                      value=""
                    />
                    <button
                      aria-label="clear"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                      style="visibility: hidden;"
                      tabindex="0"
                      title="clear"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                        data-testid="ClearIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                        />
                      </svg>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDataGrid-main css-204u17-MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnHeaders css-z8fhq1-MuiDataGrid-columnHeaders"
            style="min-height: 39px; max-height: 39px; line-height: 39px;"
          >
            <div
              class="MuiDataGrid-columnHeadersInner css-gl260s-MuiDataGrid-columnHeadersInner"
              role="rowgroup"
              style="transform: translate3d(0px, 0px, 0px);"
            >
              <div
                aria-rowindex="1"
                class="css-yrdy0g-MuiDataGrid-columnHeaderRow"
                role="row"
              >
                <div
                  aria-colindex="1"
                  aria-label="organizer.kcId"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                  data-field="kcId"
                  role="columnheader"
                  style="height: 39px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="0"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          organizer.kcId
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 39px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
                <div
                  aria-colindex="2"
                  aria-label="organizer.name"
                  aria-sort="none"
                  class="MuiDataGrid-columnHeader MuiDataGrid-columnHeader--sortable"
                  data-field="name"
                  role="columnheader"
                  style="height: 39px; width: 0px; min-width: 0; max-width: 0;"
                  tabindex="-1"
                >
                  <div
                    class="MuiDataGrid-columnHeaderDraggableContainer"
                    draggable="false"
                  >
                    <div
                      class="MuiDataGrid-columnHeaderTitleContainer"
                    >
                      <div
                        class="MuiDataGrid-columnHeaderTitleContainerContent"
                      >
                        <div
                          aria-label=""
                          class="MuiDataGrid-columnHeaderTitle css-1jbbcbn-MuiDataGrid-columnHeaderTitle"
                          data-mui-internal-clone-element="true"
                        >
                          organizer.name
                        </div>
                      </div>
                      <div
                        class="MuiDataGrid-iconButtonContainer css-ltf0zy-MuiDataGrid-iconButtonContainer"
                      >
                        <button
                          aria-label="Sort"
                          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                          tabindex="-1"
                          title="Sort"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall MuiDataGrid-sortIcon css-ptiqhd-MuiSvgIcon-root"
                            data-testid="ArrowUpwardIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiDataGrid-columnSeparator MuiDataGrid-columnSeparator--sideRight"
                    style="min-height: 39px; opacity: 1;"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                      data-testid="SeparatorIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M11 19V5h2v14z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="overflow: visible; height: 0px; width: 0px;"
          >
            <div
              class="MuiDataGrid-virtualScroller css-1w5m2wr-MuiDataGrid-virtualScroller"
              style="width: 0px; height: auto; margin-top: 39px; overflow-x: hidden;"
            >
              <div
                class="MuiDataGrid-virtualScrollerContent MuiDataGrid-virtualScrollerContent--overflowed css-1kwdphh-MuiDataGrid-virtualScrollerContent"
                style="width: auto; height: 56px; min-height: auto;"
              >
                <div
                  class="MuiDataGrid-virtualScrollerRenderZone css-s1v7zr-MuiDataGrid-virtualScrollerRenderZone"
                  style="transform: translate3d(0px, 0px, 0px);"
                >
                  <div
                    aria-rowindex="2"
                    aria-selected="false"
                    class="MuiDataGrid-row"
                    data-id="1"
                    data-rowindex="0"
                    role="row"
                    style="max-height: 28px; min-height: 28px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="kcId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 28px; max-height: 28px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      />
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="name"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 28px; max-height: 28px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                        title="Järjestäjä 1"
                      >
                        Järjestäjä 1
                      </div>
                    </div>
                  </div>
                  <div
                    aria-rowindex="3"
                    aria-selected="false"
                    class="MuiDataGrid-row MuiDataGrid-row--lastVisible"
                    data-id="2"
                    data-rowindex="1"
                    role="row"
                    style="max-height: 28px; min-height: 28px;"
                  >
                    <div
                      aria-colindex="1"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="0"
                      data-field="kcId"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 28px; max-height: 28px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                      />
                    </div>
                    <div
                      aria-colindex="2"
                      aria-colspan="1"
                      class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                      data-colindex="1"
                      data-field="name"
                      role="cell"
                      style="min-width: 0; max-width: 0; min-height: 28px; max-height: 28px;"
                      tabindex="-1"
                    >
                      <div
                        class="MuiDataGrid-cellContent"
                        title="Järjestäjä 2"
                      >
                        Järjestäjä 2
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
        <div>
          <div
            class="MuiDataGrid-footerContainer css-17jjc08-MuiDataGrid-footerContainer"
          >
            <div />
            <div
              class="MuiTablePagination-root css-rtrcn9-MuiTablePagination-root"
            >
              <div
                class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
              >
                <div
                  class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
                />
                <p
                  class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
                >
                  1–2 of 2
                </p>
                <div
                  class="MuiTablePagination-actions"
                >
                  <button
                    aria-label="Go to previous page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to previous page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowLeftIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                      />
                    </svg>
                  </button>
                  <button
                    aria-label="Go to next page"
                    class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                    disabled=""
                    tabindex="-1"
                    title="Go to next page"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="KeyboardArrowRightIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
